### PR TITLE
style: Cleanup a bit the scroll-snap-type properties.

### PR DESCRIPTION
**Pull request by emilio**
Still not perfect, but at least a bit more consistent.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19018)

### CODE REVIEW

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -3301,9 +3301,9 @@ fn static_assert() {
 
     ${impl_animation_timing_function()}
 
-    <% scroll_snap_type_keyword = Keyword("scroll-snap-type", "none mandatory proximity") %>
-
+    <% scroll_snap_type_keyword = Keyword("scroll-snap-type", "None Mandatory Proximity") %>
```
Comment (wafflespeanut) [inline components/style/properties/gecko.mako.rs]: Just curious - what's with the title cases?

File: components/style/properties/longhand/box.mako.rs [hunk 2]
```diff
@@ -1451,20 +1451,17 @@ ${helpers.single_keyword("scroll-behavior",
...
-    pub use super::scroll_snap_type_x::parse;
-</%helpers:longhand>
+% for axis in ["x", "y"]:
+    ${helpers.predefined_type(
+        "scroll-snap-type-" + axis,
+        "ScrollSnapType",
+        "computed::ScrollSnapType::None",
+        products="gecko",
+        needs_context=False,
+        spec="Nonstandard (https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-type-x)",
```
Comment (wafflespeanut) [inline components/style/properties/longhand/box.mako.rs]: Nit: different links?

File: components/style/properties/gecko.mako.rs (hunk 1 above)
Comment (emilio) [reply components/style/properties/gecko.mako.rs]: This generates a `Match` statement. All this is a mess, I want to rip it off... :)

Comment (emilio) [reply components/style/properties/gecko.mako.rs]: I didn't remember it auto-capitalized the names, huh :)

File: components/style/properties/longhand/box.mako.rs (hunk 2 above)
Comment (emilio) [reply components/style/properties/longhand/box.mako.rs]: Meh

**Comment by wafflespeanut** [comment]
LGTM, r=me :)

File: components/style/properties/longhand/box.mako.rs (hunk 2 above)
Comment (emilio) [reply components/style/properties/longhand/box.mako.rs]: (I'd rather make the link legible in the source, but if you want, I can change it)

Comment (wafflespeanut) [reply components/style/properties/longhand/box.mako.rs]: Nah, never mind...

File: components/style/properties/gecko.mako.rs (hunk 1 above)
Comment (emilio) [reply components/style/properties/gecko.mako.rs]: That is, before this, the enum variants were actually `T::none`, `T::mandatory`, `T::proximity`, now they have rusty case.

**Comment by emilio** [comment]
Thanks for taking a look @wafflespeanut!

@bors-servo r=wafflespeanut

File: components/style/properties/gecko.mako.rs (hunk 1 above)
Comment (wafflespeanut) [reply components/style/properties/gecko.mako.rs]: Yep, got it 👍

**Pull request merged**